interactions/client/get.py

```diff
@@ -0,0 +1,289 @@
+                        _request = _func(**_kwargs)
+                        _objects[_index] = _request
+                return _http_request(_obj, http=client._http, request=_objects)
+
+        _obj: Optional[_A] = None
+
+        force_cache = kwargs.pop("force", None) == "cache"
+        force_http = kwargs.pop("force", None) == "http"
+        if not force_http:
+            _obj = _get_cache(obj, client, kwarg_name, **kwargs)
+
+        if force_cache:
+            return _obj
+
+        elif not force_http and _obj:
+            return _return_cache(_obj)
+
+        else:
+            return _http_request(obj=obj, http=client._http, _name=http_name, **kwargs)
+
+    elif len(args) == 1:
+        return _search_iterable(*args, **kwargs)
+
+
+async def _http_request(
+    obj: Type[_A],
+    http: HTTPClient,
+    request: Union[Coroutine, List[Union[_A, Coroutine]], List[Coroutine]] = None,
+    _name: str = None,
+    **kwargs,
+) -> Union[_A, List[_A]]:
+
+    if not request:
+        if obj in (Role, Emoji):
+            _guild = Guild(**await http.get_guild(kwargs.pop("guild_id")), _client=http)
+            _func = getattr(_guild, _name)
+            return await _func(**kwargs)
+
+        _func = getattr(http, _name)
+        _obj = await _func
+        return obj(**_obj, _client=http)
+
+    if not isinstance(request, list):
+        return obj(**await request, _client=http)
+
+    return [obj(**await req, _client=http) if isawaitable(req) else req for req in request]
+
+
+async def _return_cache(
+    obj: Union[Optional[_A], List[Optional[_A]]]
+) -> Union[Optional[_A], List[Optional[_A]]]:
+    await sleep(0)  # iirc Bluenix meant that any coroutine should await at least once
+    return obj
+
+
+def _get_cache(
+    _object: Type[_A], client: Client, kwarg_name: str, _list: bool = False, **kwargs
+) -> Union[Optional[_A], List[Optional[_A]]]:
+
+    if _list:
+        _obj = []
+        if isinstance(_object, Member):  # Can't be more dynamic on this
+            _values = ()
+            _guild_id = Snowflake(kwargs.get("guild_id"))
+            for _id in kwargs.get("member_ids"):
+                _values += (
+                    (
+                        _guild_id,
+                        Snowflake(_id),
+                    ),
+                )
+            _obj.extend(client.cache[_object].get(item) for item in _values)
+
+        else:
+            _obj.extend(
+                client.cache[_object].get(Snowflake(_id), None) for _id in kwargs.get(kwarg_name)
+            )
+    else:
+        if isinstance(_object, Member):
+            _values = (
+                Snowflake(kwargs.get("guild_id")),
+                Snowflake(kwargs.get("member_id")),
+            )  # Fuck it, I can't be dynamic on this
+        else:
+            _values = Snowflake(kwargs.get(kwarg_name))
+
+        _obj = client.cache[_object].get(_values)
+    return _obj
+
+
+def _search_iterable(*args, **kwargs) -> Optional[_A]:
```

Comment:
`_search_iterable` only ever gets a single argument, why not show that in the parameters?

Reply:
Same thing applies to the keyword arguments. Double stars doesn't let you set a parameter name, which makes it confusing to know what to pass when first looking at it

Reply:
It takes `**kwargs` because you can put any attribute in it, setting a value to it doesn't allow that. I'll change the `*args` tho